crates/thread_aware/src/cell.rs

```diff
@@ -422,7 +422,10 @@ impl<T, S: Strategy> Trc<T, S> {
 
 impl<T, S: Strategy> ThreadAware for Trc<T, S> {
     fn relocated(self, source: MemoryAffinity, destination: MemoryAffinity) -> Self {
-        let value = self.storage.read().expect("Failed to acquire read lock").get_clone(destination);
+        let value = match destination {
```

Comment:
This implies that relocating to an unknown destination creates a new value. I don't think that's what we want? That will result in potentially unbounded number of connection pools for HTTP client for example.

Do we really need to have unknown affinity for the destination as well? Couldn't we get away with just having it for source?

More generally, I suspect the unknown affinity will hurt in other places as well, which is why I'd prefer not to introduce it and rather embrace the "you only get one Builtins outside of the runtime" approach.

Reply:
I agree with the changing destination to never be a unknown affinity; Until we make the final decision on the runtime design I'm going to keep the concept of MemoryAffinity::Unknown for the source however.

Reply:
Yeah, one "option" we discussed was having an `Option<Affinity>` for the source instead. For source we need it, but I agree destination probably not. 

> This implies that relocating to an unknown destination creates a new value. I don't think that's what we want? That will result in potentially unbounded number of connection pools for HTTP client for example.

What I don't get is, why would that imply it, we could just implement "outside" as always a single, shared affinity?

Reply:
> What I don't get is, why would that imply it, we could just implement "outside" as always a single, shared affinity?
My comment was pointed at the current implementation at the time.

On a more general note, even the "single outside affinity" has issues - how do you relocate e.g. IoDriver to it? IoDriver is tied to a thread that drives the loop, so relocating it outside would be problematic.